pass/ProteusPass.cpp

```diff
@@ -497,19 +572,10 @@ struct ProteusJitPassImpl {
     Function *JITFn = JITInfo.first;
     JitFunctionInfo &JFI = JITInfo.second;
 
-    // Create jit entry runtime function.
-    Type *VoidPtrTy = Type::getInt8PtrTy(M.getContext());
-    Type *Int32Ty = Type::getInt32Ty(M.getContext());
-    Type *Int128Ty = Type::getInt128Ty(M.getContext());
-    // Use Int64 type for the Value, big enough to hold primitives.
-    StructType *RuntimeConstantTy =
-        StructType::create({Int128Ty}, "struct.args");
-
-    FunctionType *JitEntryFnTy =
-        FunctionType::get(VoidPtrTy,
-                          {VoidPtrTy, VoidPtrTy, Int32Ty,
-                           RuntimeConstantTy->getPointerTo(), Int32Ty},
-                          /* isVarArg=*/false);
+    FunctionType *JitEntryFnTy = FunctionType::get(
+        PtrTy,
+        {PtrTy, PtrTy, Int32Ty, RuntimeConstantTy->getPointerTo(), Int32Ty},
+        /* isVarArg=*/false);
```

Comment:
Create a function in `ProteusJitPassImpl` (`getOrInsertProteusEntry`) that creates both the FnTy and the FunctionCallee.

Reply:
You suggestion is good but I want to address later encapsulating interface function creation. Can you open an issue?